File: Códigos/HackerRank/Recursive_Digit_Sum.c
/**
 * Problem: Recursive Digit Sum
 * url: https://www.hackerrank.com/challenges/recursive-digit-sum/problem
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define MAX 15

int super_digit(long int p) {
	char p_char[MAX];
	int sum = 0;
	
	sprintf(p_char, "%ld", p);
	
	if (strlen(p_char) == 1)
		return atoi(p_char);
	
	for (int i=0; i<strlen(p_char); i++) {
		sum += p_char[i] - '0';
	}
	
	return super_digit(sum);
}


int main(int argc, char const *argv[]) {
	char digit;
	int k;
	long int p;
	
	while(scanf("%c", &digit), digit != ' ') {
		p += digit-'0';
	}
	
	scanf("%d", &k);
	
	p *= k;
	
	printf("%ld", super_digit(p));
	
	return 0;
}

File: Códigos/HackerRank/Recursive_Digit_Sum.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

""" 
    Problem: Recursive Digit Sum
    url: https://www.hackerrank.com/challenges/recursive-digit-sum/problem
"""

import sys


def read_character():
	return sys.stdin.read(1)
	

def read_line():
	return sys.stdin.readline().strip()
	

def super_digit(p):
	if len(str(p)) == 1:
		return p
	
	sum = 0
	for digit in str(p):
		sum += int(digit)
	
	return super_digit(sum)
		


def main():
	sum = 0
	
	while(True):
		digit = read_character()
		if digit == ' ':
			break
		sum += int(digit)
	
	k = int(read_line())
	sum *= k
	
	print(super_digit(sum))


if __name__ == '__main__':
    main()

